js/components/graph/Graph.js

```diff
@@ -641,7 +647,7 @@ export default class Graph extends React.Component {
 
     return (
       <div id="react-graph" className="react-graph" onClick={this.props.closeSidebar}>
-        <CourseModal ref={this.modal} />
+        {this.state.showCourseModal && <CourseModal courseChange={this.state.courseId} callbackModal={this.callbackModal} />}
```

Comment:
Rather than conditionally render the `CourseModal` component, It's better to pass the `showCourseModal` as a prop that controls whether it appears or not. This prop replaces the `modalIsOpen` state from the `CourseModal` component.

Also, don't think of the course code as a "change", it's just what course that's being shown for the modal. Again, this prop should replace one of the states of the `CourseModal` component.